os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -563,6 +564,25 @@ void binary_manager_release_binary_sem(int bin_idx)
 		}
 		sem = sq_next(sem);
 	}
+
+	/* Release all holders for user sempahores */
+	sem = (sem_t *)sq_peek(&BIN_SEMLIST(bin_idx));
+	while (sem) {
+#if CONFIG_SEM_PREALLOCHOLDERS > 0
+		for (holder = sem->hhead; holder; holder = holder->flink)
+#else
+		holder = &sem->holder;
+#endif
+		{
+			if (holder) {
+				/* Free a holder of semaphore */
+				sem_freeholder(sem, holder);
+			}
+		}
+		sem = sq_next(sem);
+	}
+	sq_init(&BIN_SEMLIST(bin_idx));
```

Comment:
You only initialize the sem list in here. Is it not needed to initialize before using?

Reply:
It is ok because of global variable but I will update.